styled/segment.go

```diff
@@ -0,0 +1,104 @@
+package styled
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/elves/elvish/eval/vals"
+)
+
+// A styled Segment is a string that has some style applied to it.
```

Comment:
Docstrings for exported symbols should start with the symbol itself (https://golang.org/doc/effective_go.html#commentary). In this case, simply remove "A styled", which is apparent from the package it is in.

Reply:
Alright I will rework and add missing docstrings.